jina/helper.py

```diff
@@ -1684,3 +1684,34 @@ def _telemetry():
             pass
 
     threading.Thread(target=_telemetry, daemon=True).start()
+
+
+_DEFAULT_GRPC_OPTION = [
+    ('grpc.max_send_message_length', -1),
+    ('grpc.max_receive_message_length', -1),
+    # for the following see this blog post for the choice of default value https://cs.mcgill.ca/~mxia3/2019/02/23/Using-gRPC-in-Production/
+    ('grpc.keepalive_time_ms', 10000),
+    # send keepalive ping every 10 second, default is 2 hours.
+    ('grpc.keepalive_timeout_ms', 5000),
+    # keepalive ping time out after 5 seconds, default is 20 seconds
+    ('grpc.keepalive_permit_without_calls', True),
+    # allow keepalive pings when there's no gRPC calls
+    ('grpc.http2.max_pings_without_data', 0),
+    # allow unlimited amount of keepalive pings without data
+    ('grpc.http2.min_time_between_pings_ms', 10000),
+    # allow grpc pings from client every 10 seconds
+    ('grpc.http2.min_ping_interval_without_data_ms', 5000),
+    # allow grpc pings from client without data every 5 seconds
+]
+
+
+def _get_grpc_server_options(option_from_args: Dict) -> List[Tuple[str, Any]]:
+
+    if option_from_args:
+        options_from_dict = []
```

Comment:
Shouldn't this also be filled with all options in `_DEFAULT_GRPC_OPTION` that are not given in `option_from_args`? Right now it looks like as soon as the user specifies one option in `option_from_args`, all of our defaults will be ignored.

Reply:
Yes I had the same thinking.

The things is I don't like than the stuff happen under the hoods from the users. Only advanced users will use it anyway so I will assume that they know what they are doing and prefer this behavior.

What do you think @alaeddine-13 ?

Reply:
Actually I think grouping grpc parameters into the same grpc options argument does not go well with the pattern we follow:
we do not group parameters for docker or executor,...
But in case we opt for grouping parameters in the same argument, I'd go with @JohannesMessner , I find it more intuitive

Reply:
Yeah but I can't expose all of the possible grpc parameters in the Flow right ? What would you suggest ?
The things is that we don't usually let users do override our parameters

Reply:
I also wouldn't expose all possible grpc arguments independently. I see this similarly to how you just pass an entire config to the storage backend in docarray, and what you don't precise gets filled in with the defaults. I would do the same here.